src/extensions/replay/triggerMatching.ts

```diff
@@ -0,0 +1,314 @@
+import {
+    SESSION_RECORDING_EVENT_TRIGGER_ACTIVATED_SESSION,
+    SESSION_RECORDING_URL_TRIGGER_ACTIVATED_SESSION,
+} from '../../constants'
+import { PostHog } from '../../posthog-core'
+import { FlagVariant, RemoteConfig, SessionRecordingUrlTrigger } from '../../types'
+import { isBoolean, isObject, isString } from '../../utils/type-utils'
+import { isNullish } from '../../utils/type-utils'
+import { window } from '../../utils/globals'
+
+export type TriggerType = 'url' | 'event'
+export type TriggerStatus = 'trigger_activated' | 'trigger_pending' | 'trigger_disabled'
```

Comment:
Could you, pretty please, add the same description as you did for the session recording status right below? It would clarify lots of logic.

My bet would be something like this, but not sure if it's correct :) 

```ts
/**
 * Trigger status indicates the current state of a trigger in the session recording system.
 * - trigger_activated: The trigger condition has been met and recording is active for this session
 * - trigger_pending: The trigger is configured but hasn't been activated yet for this session
 * - trigger_disabled: No trigger configuration exists or the trigger is explicitly disabled
 */
 ```